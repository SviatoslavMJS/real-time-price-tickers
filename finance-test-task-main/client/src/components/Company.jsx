import s from '../styles/Company.module.css';

function Company(props) {
    return (
        <div style={{ backgroundColor: props.color }} className={s.ticker}>
            {props.ticker}
        </div>
    )
}

export default Company;