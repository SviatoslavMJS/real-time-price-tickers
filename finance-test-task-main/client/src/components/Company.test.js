import Company from "./Company.jsx";
import renderer from 'react-test-renderer';

test("On change prop company in Company text must change from APPL to SSSR", () => {
    const component = renderer.create(
      <Company color={"#666666"} company={"APPL"}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.company = "SSSR";

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.company = "APPL";

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
});

test("On change prop color in Company style color must be changed from #666666 to #0d0a08", () => {
    const component = renderer.create(
      <Company color={"#666666"} company={"APPL"}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.company = "#0d0a08";

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.company = "#666666";

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
});
