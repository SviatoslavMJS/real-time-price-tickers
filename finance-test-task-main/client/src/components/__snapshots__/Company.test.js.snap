// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On change prop color in Company style color must be changed from #666666 to #0d0a08 1`] = `
<div
  className="ticker"
  style={
    Object {
      "backgroundColor": "#666666",
    }
  }
/>
`;

exports[`On change prop color in Company style color must be changed from #666666 to #0d0a08 2`] = `
<div
  className="ticker"
  style={
    Object {
      "backgroundColor": "#666666",
    }
  }
/>
`;

exports[`On change prop color in Company style color must be changed from #666666 to #0d0a08 3`] = `
<div
  className="ticker"
  style={
    Object {
      "backgroundColor": "#666666",
    }
  }
/>
`;

exports[`On change prop company in Company text must change from APPL to SSSR 1`] = `
<div
  className="ticker"
  style={
    Object {
      "backgroundColor": "#666666",
    }
  }
/>
`;

exports[`On change prop company in Company text must change from APPL to SSSR 2`] = `
<div
  className="ticker"
  style={
    Object {
      "backgroundColor": "#666666",
    }
  }
/>
`;

exports[`On change prop company in Company text must change from APPL to SSSR 3`] = `
<div
  className="ticker"
  style={
    Object {
      "backgroundColor": "#666666",
    }
  }
/>
`;
